### DISCUSSION TITLE
Homebrew installation stuck

### DISCUSSION THREAD

**Discussion by krishnakumarv1** [Homebrew/discussions#3053] [Getting started]
Hi, I am a noob here trying to get past this while trying to install Homebrew on my Intel Macbook Pro. Can someone help?

```
==> /usr/bin/sudo /usr/sbin/chown -R username:admin /usr/local/Homebrew
==> Searching online for the Command Line Tools
==> /usr/bin/sudo /usr/bin/touch /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress
```

Thank you in advance!

**Comment by SMillerDev** [comment]
It's installing the Apple commandline tools. You can either try installing them yourself beforehand using `xcode-select --install` or you can just wait while brew installs them.

**Reply by sergeidmitri** [reply]
Did anyone’s battery get fried after installing Homebrew?

**Comment by krishnakumarv1** [comment]
I waited for a few hours; my Apple Software update process was also not
responding. I checked my Brew status using another terminal window and it
seemed to be working. So I closed the Terminal window, where Homebrew
installation was stuck, and I used 'brew doctor', 'brew upgrade', and 'brew
cleanup.' So far things look okay.

**Reply by carlocab** [reply]
What's the output of `brew config`?

**Reply by Danielg212** [reply]
>➜  ~ brew config
```Traceback (most recent call last):
	11: from /usr/local/Homebrew/Library/Homebrew/brew.rb:31:in `<main>'
	10: from /usr/local/Homebrew/Library/Homebrew/brew.rb:31:in `require_relative'
	 9: from /usr/local/Homebrew/Library/Homebrew/global.rb:80:in `<top (required)>'
	 8: from /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 7: from /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/2.6.3_2/lib/ruby/2.6.0/rubygems/core_ext/kernel_require.rb:54:in `require'
	 6: from /usr/local/Homebrew/Library/Homebrew/os.rb:7:in `<top (required)>'
	 5: from /usr/local/Homebrew/Library/Homebrew/os.rb:43:in `<module:OS>'
	 4: from /usr/local/Homebrew/Library/Homebrew/os/mac.rb:60:in `prerelease?'
	 3: from /usr/local/Homebrew/Library/Homebrew/os/mac.rb:28:in `version'
	 2: from /usr/local/Homebrew/Library/Homebrew/os/mac/version.rb:33:in `from_symbol'
	 1: from /usr/local/Homebrew/Library/Homebrew/os/mac/version.rb:33:in `fetch'
/usr/local/Homebrew/Library/Homebrew/os/mac/version.rb:33:in `block in from_symbol': unknown or unsupported macOS version: :dunno (MacOSVersionError)```

**Reply by gromgit** [reply]
Run `brew update-reset`, then `brew config` again.

**Reply by krishnakumarv1** [reply]
This is what I see.

```
HOMEBREW_VERSION: 3.4.1
ORIGIN: https://github.com/Homebrew/brew
HEAD: ea0ddc3297739b7380887d4a7e6b6590c314dca9
Last commit: 17 hours ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 7680186d03369679d2ea6d605eca32fc7374a2d1
Core tap last commit: 10 hours ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_CORE_GIT_REMOTE: https://github.com/Homebrew/homebrew-core
HOMEBREW_MAKE_JOBS: 12
Homebrew Ruby: 2.6.8 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: dodeca-core 64-bit kabylake
Clang: 13.0.0 build 1300
Git: 2.35.1 => /usr/local/bin/git
Curl: 7.77.0 => /usr/bin/curl
macOS: 12.2.1-x86_64
CLT: 13.2.0.0.1.1637719136
Xcode: N/A
```

Sorry for the delay in my response.

**Reply by gromgit** [reply]
Since `brew config` doesn't throw an error now, run `brew doctor` to be sure, and fix whatever it tells you to fix, then your Homebrew installation should be good to go.

**Comment by htpusa** [comment]
Hi,
It seems I have this exact same problem. OS X 12.2.1.
It might be a Mac issue: when trying to install command line tools from the prompt that comes up if you try git, the same thing happens, it's forever searching for the software.

**Comment by Calorion** [comment]
I was able to fix _my_ problem: On Big Sur, the latest version of Xcode will not install. Downloading and installing [Command Line Tools for Xcode 12.5.1.dmg](https://download.developer.apple.com/Developer_Tools/Command_Line_Tools_for_Xcode_12.5.1/Command_Line_Tools_for_Xcode_12.5.1.dmg) fixed the problem.

**Reply by LegalInspiration** [reply]
Please note, this absolutely works, but you must be signed in on the Apple Developer site or the link given will not open - you will get an authorization error.

If you aren't signed in, go to this link: [https://developer.apple.com/xcode/resources/](https://developer.apple.com/xcode/resources/ )

and sign in with your AppleID. It will ask you to agree to the terms and conditions. Once you do, the link above will produce a .dmg with the command line tools.

**Comment by ScottS-byte** [comment]
Same issues here. 2021 MBP M1 and brew stalled every other install, or more frequently. Some of the smaller packages, like `tree`, succeed, but bigger packages, like `graphviz` failed miserably.

There really isn't much to say. I created a brew bundle to migrate from an Intel-based MBP, and ended up having to install each brew package individually, many times hitting command line up arrow to try again, and try again... I've never seen brew behave so poorly in the past.

Ran brew doctor, even completely removed brew two times. Ug.

**Comment by CehKah** [comment]
Hi, folks, being new to macs but also being stubborn, I resolved the above mentioned problem by killing the command line tools update process (ctrl+c) and then started the installation all over again WITHOUT removing the files from the first attempt. Worked for me, cheers CehKah
